.gitignore

```diff
@@ -54,6 +54,9 @@ environment.y*
 *.gch
 coverage/
 coverage.info
+*.h5
+*.csv
+/*.yaml
```

Comment:
I don't think we should ignoring `.csv` files by default, given that there CSV files that are properly part of the repository.

I'm a little ambivalent about excluding `.h5` and `.yaml` files in the root directory. I guess you end up with these the same way I do, by running some of the Python examples directly from the repository root. But I'd rather this be visible as clutter that I should delete rather than just teaching git to ignore it (you can always add these to your local `.git/info/exclude`).

Reply:
I don't have a problem dropping the `.gitignore` portion.

And you're right - I end up with these files all the time while testing.